### PR TITLE
Pull the fully qualified cub:: namespace into thrust::

### PR DESCRIPTION
**Pull request by allisonvacanti**
This will allow us to just use `cub::` instead of `CUB_NS_QUALIFIER` as long as we're in `thrust::`.

This effectively reverts a large portion of 363c352 and makes it less obnoxious to use CUB from Thrust.

### CODE REVIEW

**Comment by allisonvacanti** [comment]
DVS CL: 30207484

run tests

**Comment by allisonvacanti** [comment]
DVS CL: 30207526

run tests

**Review by senior-zero — APPROVED** [review]
Here is the new approach overview:

```cpp
namespace cub {
void g() { }
}

namespace thrust {
namespace cub {
using namespace ::cub;
}
}

namespace cub {
void k() { }
}

namespace thrust {
void f() {
  cub::g();
  cub::k();
}
}

int main() {
  thrust::f();
}
```

I haven't known that it's possible to extend a namespace [after using directive](https://en.cppreference.com/w/cpp/language/namespace#Using-directives). So we don't rely on headers order and have readable code, nice!

**Comment by allisonvacanti** [comment]
It also lets us avoid having to use the `CUB_NS_QUALIFIER` macro from Thrust just to cover the edgecase where `THRUST_WRAPPED_NAMESPACE` and `CUB_WRAPPED_NAMESPACE` are different :)

**Comment by allisonvacanti** [comment]
DVS shows some regressions in cuSPARSE builds with this patch. Investigating...

**Comment by allisonvacanti** [comment]
Added a default definition of the new `CUB_NS_QUALIFIER` macro when `THRUST_IGNORE_CUB_VERSION_CHECK` is defined. Restarting tests:

DVS CL: 30215528

run tests

**Comment by allisonvacanti** [comment]
DVS is still failing, cusparse uses an old copy of CUB that defines `CUB_NS_[PREFIX|POSTFIX]` but not `CUB_NS_QUALIFIER`. I've contacted their maintainers to help resolve this.

**Comment by allisonvacanti** [comment]
Resubmitted DVS under CL 30226678

**Comment by allisonvacanti** [comment]
DVS CL: 30227336

run tests

**Comment by allisonvacanti** [comment]
New DVS CL: 30228004

run tests

**Comment by allisonvacanti** [comment]
The cusparse patch introduced another issue. Working with that team to resolve it before merging.

**Pull request merged**
